http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/api/ProxyRoutes.scala

```diff
@@ -237,17 +256,35 @@ class ProxyRoutes(proxyService: ProxyService, corsSupport: CorsSupport, refererC
     for {
       ctx <- ev.ask[AppContext]
       apiCall = proxyService.proxyRequest(userInfo, googleProject, runtimeName, request)
-      _ <- metrics.incrementCounter("proxyRuntime")
       resp <- ctx.span.fold(apiCall)(span => spanResource[IO](span, "proxyRuntime").use(_ => apiCall))
+
+      tool = RuntimeContainerServiceType.values
+        .find(s => request.uri.toString.contains(s.proxySegment))
+        .map(_.imageType.entryName)
+        .getOrElse("other")
+
+      _ <- if (request.uri.toString.endsWith(".ipynb") && request.method == HttpMethods.PUT) {
+        request.entity.contentLengthOption.traverse(size => metrics.gauge("proxy/notebooksSize", size.toDouble))
```

Comment:
going forward we want to keep metric names to a single string (with no additional `/`s) after `OpenCensus/leonardo/` so that we have a uniform naming pattern. We can think about extracting as much as possible into tags so maybe for this metric name, we can use `notebookSize` with tag `source: proxy` or something like that. then going forward we can add other notebook attribute metrics with same or different sources